tests/test_basic_process.cpp

```diff
@@ -38,7 +38,7 @@ TEST_F(BasicProcessTest, execute_missing_command)
     EXPECT_FALSE(process_state.completed_successfully());
     EXPECT_FALSE(process_state.exit_code);
 
-    EXPECT_TRUE(process_state.error);
+    ASSERT_TRUE(process_state.error);
```

Comment:
I'm not sure I understand why this was needed. Yes, `error` is a `std::optional`, and we want to check if it has a value. But what does the assert do differently from the expect?

Reply:
Assert will abort the test so if there is no value in the optional we don't crash from trying to access it

Reply:
ah, i wasn't reading the line just below this. apologies.
but this raises the question: if the error optional does not have a value, why do we assert/expect it to have one. is the content of `error` not deterministic? shouldn't we have the test always pass?

also, not requesting this as a change, but what's your opinion on something like
```cpp
const auto process_has_error = process_state.error.has_value();
EXPECT_TRUE(process_has_error);
if (process_has_error) {
    EXPECT_EQ(..., process_state.error->state);
}
```
I'm asking this because I think (please correct me) that if an EXPECT fails, the test will report all of the other failed EXPECTs, but if an ASSERT fails, the test won't report any other failed EXPECTs.

Reply:
The test is making sure that the optional does have a value, if the unit under test has bug then the optional might not have a value and the test should fail. Before this PR the unit had a bug and instead of failing it would crash. 

I think that change would be good if we had more assets/expects after we check the value, assuming those checks make sense if there is no value. ASSERT will abort the test so any other expects after won't be reported, but any EXPECT before the ASSERT will still be reported

Reply:
Ok, perfect. Makes sense. Thanks for the details!